src/cogent3/core/new_alignment.py

```diff
@@ -2666,17 +2785,31 @@ def get_gapped_seq_array(
         start: OptInt = None,
         stop: OptInt = None,
     ) -> numpy.ndarray:
-        """Return sequence corresponding to seqid as an array of indices.
+        """Return sequence data corresponding to seqid as an array of indices.
         start/stop are in alignment coordinates. Includes gaps.
         """
-        seq = self.seqs[seqid]
-        gaps = self.gaps[seqid]
+        seq = self._seqs[seqid]
 
         # if there's no gaps, just slice the sequence
         if len(seq) == self.align_len:
             return seq[start:stop]
 
-        return self.moltype.most_degen_alphabet().gapped_by_map(seq, gaps)[start:stop]
+        gaps = self._gaps[seqid]
```

Comment:
add a comment "refactor: design". The iter over spans will be slower than a more direct numpy based approach, but it works!